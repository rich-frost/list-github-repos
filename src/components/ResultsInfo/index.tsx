import { Center, Text, Badge } from '@chakra-ui/react';

type ResultsInfoProps = {
  count: number;
  topic: string;
};

export default function ResultsInfo({ count, topic }: ResultsInfoProps) {
  const badgeColour = count > 0 ? 'green' : 'red';
  return (
    <Center p={2}>
      <Text>
        <Badge colorScheme={badgeColour}>{count.toLocaleString()}</Badge>{' '}
        respositories found for topic <Text as="b">{topic}</Text>
      </Text>
    </Center>
  );
}
