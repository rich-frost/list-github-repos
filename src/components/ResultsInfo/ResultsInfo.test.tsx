import renderer from 'react-test-renderer';
import ResultsInfo from './index';

describe('ResultsInfo component', () => {
  it('renders the component', () => {
    const component = renderer.create(
      <ResultsInfo count={123} topic="React" />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
