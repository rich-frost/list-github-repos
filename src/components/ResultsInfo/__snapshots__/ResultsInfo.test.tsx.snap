// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsInfo component renders the component 1`] = `
<div
  className="css-1w3riw2"
>
  <p
    className="chakra-text css-0"
  >
    <span
      className="chakra-badge css-ifsvay"
    >
      123
    </span>
     
    respositories found for topic 
    <b
      className="chakra-text css-0"
    >
      React
    </b>
  </p>
</div>
`;
